File: express/src/controller/news-controller.js
import newsService from "../service/news-service.js";
import { response } from "../response/response.js";

// Controller untuk mendapatkan komentar berita berdasarkan ID
const getNewsCommentsById = async (req, res, next) => {
  try {
    const result = await newsService.getNewsCommentsById(req.params.id);
    response(200, result, "Get news comment success", res);
  } catch (error) {
    next(error);
  }
};

// Controller untuk test, hanya mengembalikan data statis
const test = async (req, res, next) => {
  try {
    res.status(200).json({
      data: "test",
    });
  } catch (error) {
    next(error);
  }
};

export default {
  getNewsCommentsById,
  test,
};

File: flask/src/controller/news_controller.py
from response.response import response
from response.response_error import ResponseError
from service.news_service import get_news_comments

# Controller untuk menangani request dan mengembalikan komentar berita berdasarkan ID
def get_news_comments_controller(id):
    try:
        result = get_news_comments(id)
        return response(200, result, "Get news comments success")
    except ResponseError as e:
        raise e